### PR TITLE
Add sanctioned towns hover component to nation status screen

### PR DESCRIPTION
**Pull request by Fruitloopins**
#### Description: 

This adds a "[Sanctioned Towns]" section to the end of the nation status screen, it acts just like the enemies, allies, towns etc. hover components, clicking runs the list command (pictured below)

image
____
#### New Nodes/Commands/ConfigOptions: 

N/A
____
#### Relevant Towny Issue ticket:

N/A
____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java [hunk 1]
```diff
@@ -507,6 +507,18 @@ public static StatusScreen getStatus(Nation nation, CommandSender sender) {
...
+		List<String> sanctionedTowns = getFormattedNames(nation.getSanctionedTowns());
+		if (sanctionedTowns.size() > 10)
+			shortenOverLengthList(sanctionedTowns, 11, translator);
+		
+		if (sanctionedTowns.size() > 0)
+			screen.addComponentOf("sanctionedtowns", colourHoverKey(translator.of("status_nation_sanctioned_towns")),
+				HoverEvent.showText(TownyComponents.miniMessage(getFormattedStrings(translator.of("status_nation_sanctioned_towns"), sanctionedTowns, nation.getSanctionedTowns().size()))
+					.append(Component.newline())
+					.append(translator.component("status_hover_click_for_more"))),
+				ClickEvent.runCommand("/towny:nation sanctiontown list"));
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java:521]: This should also take in the nation name so you're not only seeing your own nation's sanctioned towns.

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java (hunk 1 above)
Comment (Fruitloopins) [reply Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java:521]: I'll fix it when I wake up

**Comment by Fruitloopins** [comment]
should be all good now i think, i added the nation name to the clickevent

File: Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java [hunk 2]
```diff
@@ -507,6 +507,18 @@ public static StatusScreen getStatus(Nation nation, CommandSender sender) {
 						.append(Component.newline())
 						.append(translator.component("status_hover_click_for_more"))),
 				ClickEvent.runCommand("/towny:nation enemylist " + nation.getName()));
+		
+		// Sanctioned Towns [3]: Prague, Berlin, Vienna
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java:511]: This comment should look like so: `// [Sanctioned Towns] + hover` (Yes I know most of the other comments are incorrect.)

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java (hunk 2 above)
Comment (Fruitloopins) [reply Towny/src/main/java/com/palmergames/bukkit/towny/TownyFormatter.java]: done

**Review by LlmDl — APPROVED** [review]

**Pull request merged**